library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java

```diff
@@ -154,6 +157,13 @@ public void onCreate() {
         mExecutor = factory.createExecutor();
     }
 
+    private DownloadClientReadyChecker getDownloadClientReadyChecker() {
+        if (!(getApplication() instanceof DownloadClientReadyChecker)) {
```

Comment:
yeah @amlcurran we did look. We don't want to do it reflectively and this service can be started from a broadcast receiver when the only thing that is available is the Application.